refarch-integrations/refarch-s3-integration/refarch-s3-integration-client/src/main/java/de/muenchen/refarch/s3/integration/client/service/FileService.java

```diff
@@ -0,0 +1,149 @@
+package de.muenchen.refarch.s3.integration.client.service;
+
+import de.muenchen.refarch.s3.integration.client.exception.NoFileTypeException;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Objects;
+import java.util.stream.Collectors;
+import lombok.Getter;
+import org.apache.tika.Tika;
+import org.apache.tika.mime.MimeType;
+import org.apache.tika.mime.MimeTypeException;
+import org.apache.tika.mime.MimeTypes;
+import org.springframework.util.unit.DataSize;
+
+/**
+ * A service class for managing file extensions, detecting file types and checking file size.
+ */
+public class FileService {
+
+    private static final String NO_FILE_EXTENSION = "No file extension found for %s";
+
+    /**
+     * Map stores supported file extensions and their corresponding MIME types. If it is empty, all
+     * types are supported.
+     */
+    private final Map<String, String> supportedFileExtensions;
```

Comment:
Could you use the SupportedFileExtensions type here? If not used, why was it created?